### DISCUSSION TITLE
Code Security (Sarif) UX is painful

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#34137] [Code Security]
## Steps
1. Visit https://github.com/check-spelling/ToBeReviewedBot/security
2. Click [View alerts](https://github.com/check-spelling/ToBeReviewedBot/security/code-scanning)
    <img width="710" alt="image" src="https://user-images.githubusercontent.com/2119212/192163518-fa1ec1a5-b1e8-479c-ab9f-a082622899b5.png">
3. Ok, this branch hasn't been scanned yet, that's great, but there's a branch that has been scanned, recently, so, let's try clicking on the branch name:
4. Click [spell-check-with-spelling](https://github.com/check-spelling/ToBeReviewedBot/tree/spell-check-with-spelling)
5. Oh, it takes me to the branch, well, that's utterly unhelpful.
6. What about clicking the workflow? [Check Spelling](https://github.com/check-spelling/ToBeReviewedBot/actions/runs/3123343085)
7. Ok, that takes you to the workflow run, but you can't get to the code analysis reports
8. Maybe click [ db7b7d3](https://github.com/check-spelling/ToBeReviewedBot/commit/db7b7d363dbbb29612a450956b04cb93dccdb1ef) from there?
9. Nope, it isn't there either
10. Ok, let's go back to https://github.com/check-spelling/ToBeReviewedBot/security/code-scanning
11. Click the Tool drop down:
    <img width="935" alt="image" src="https://user-images.githubusercontent.com/2119212/192163624-9df8ef5b-f73f-457b-8062-70c2d15506b3.png">
12. Ok, so, there's a tool, and it has 0 items? What?
13. Maybe click Branch?
    <img width="853" alt="image" src="https://user-images.githubusercontent.com/2119212/192163639-54707fc1-f0b9-487e-968a-eaac06b111db.png">
14. Ok, there are no numbers here, unlike the Tool dropdown. Why?
15. Click Rule?
    <img width="803" alt="image" src="https://user-images.githubusercontent.com/2119212/192163653-3aeebf2e-b87c-4f30-90f9-f21a88275fc2.png">
16. Ok, there are rules, again, no numbers.
17. Click the first rule (forbidden pattern) -- https://github.com/check-spelling/ToBeReviewedBot/security/code-scanning?query=is%3Aopen+branch%3Aplaceholder+rule%3Aforbidden-pattern%2Cunrecognized-spelling:
    <img width="709" alt="image" src="https://user-images.githubusercontent.com/2119212/192163671-97639395-fbf4-4e81-8668-a137dad8ec2e.png">
18. Ok, we still haven't found any results.
19. Add the other rule? https://github.com/check-spelling/ToBeReviewedBot/security/code-scanning?query=is%3Aopen+branch%3Aplaceholder+rule%3Aforbidden-pattern%2Cunrecognized-spelling
    <img width="703" alt="image" src="https://user-images.githubusercontent.com/2119212/192163697-b1a94ca9-06c3-45d6-b9cd-465e2edb34e7.png">
20. No dice.
21. Note that you can click multiple rules, but if you click Branch and select `main`, you lose `placeholder` -- why?
22. Note that you can't search w/o a filter for branch -- this is in direct contradiction to how the [Actions](https://github.com/check-spelling/ToBeReviewedBot/actions) view behaves -- why?

The lack of discoverability for this thing is really painful. It turns out that I actually had a working action a month ago but never discovered that it was working because I didn't see any signs that it worked.

If I have a pull request, things work a _bit_ better, but...

## Expected

1. Clicking a branch listed at the top as having alerts should take me to the filter view of the current page
2. It should be possible to not filter to a specific branch
3. It probably should be possible to filter to things on any of a couple of branches (as one can w/ rules, or severities)
4. Numbers should be available for more than just the first dropdown
5. It'd be nice if numbers had two numbers where the second is "grand total (if nothing filtered things out beyond the current item)" -- So if I had 2 tools and one contributed 3 things and one contributed 2 things, and I happened to look at a branch where the first contributed 1 item and the second 0, I'd see (1 / 3) and (0 / 2). 

I'm sure if I figure out how to use this, I'll have more feedback.

**Comment by AlonaHlobina** [comment]
Thank you for the feedback, @jsoref!

**Comment by jsoref** [comment]
## checks view

https://github.com/check-spelling/Sanmill/pull/1/checks?check_run_id=8756077146

> 17 new alerts including 16 high severity security vulnerabilities
Warning: The base analysis for some categories could not be found. All their alerts have been reported as new.
New alerts
Security Alerts:
16 high
Other Alerts:
1 warning
See annotations below for details.

I have a number of concerns:
1. I think that `high` comes from my `precision` field and doesn't actually refer to a **severity**.
2. My code tags are `source-code` and `code-reviews`, not **Security**, so calling them **Security** alerts is misleading.

https://github.com/check-spelling/Sanmill/actions/runs/3201475847/jobs/5229495375#step:2:809
https://github.com/check-spelling/check-spelling/blob/prerelease/unknown-words.sh#L1609-L1664

3. In general, I'd really like to be able to contextualize my report.
At present, that's available here, but I don't think there's any way for me to present a link to that within the checks view: https://github.com/check-spelling/Sanmill/actions/runs/3201475847#summary-8756069787 -- Normally I'd use a comment for this, and I'd rely on annotations for the rest...

**Reply by jsoref** [reply]
The definitions here appear to agree with my understanding:
https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#reportingdescriptor-object

Name | Description
-- | --
`properties.precision` | Recommended. A string that indicates how often the results indicated by this rule are true. For example, if a rule has a known high false-positive rate, the precision should be low. Code scanning orders results by precision on GitHub so that the results with the highest level, and highest precision are shown first. Can be one of: very-high, high, medium, or low.
`properties.problem.severity` | Recommended. A string that indicates the level of severity of any alerts generated by a non-security query. This, with the properties.precision property, determines whether the results are displayed by default on GitHub so that the results with the highest problem.severity, and highest precision are shown first. Can be one of: error, warning, or recommendation.

But it seems that the doc writers and ui developers did not talk to each other about this.
